### PR TITLE
fully featured postgres connector 

### PR DESCRIPTION
**Pull request by wbittles**
While there is still work outstanding for the testing infrastructure , this drop of code runs cleanly against various changing databases. This code also includes all the required functionality and is ready for review.

### CODE REVIEW

**Comment by wbittles** [comment]
corrupt build

File: build.gradle [hunk 1]
```diff
@@ -46,12 +51,11 @@ allprojects {
     version = '2.9-SNAPSHOT'
```
Comment (planetf1) [inline build.gradle:51]: seperate issue - but note we are now on the 2.10 code streams

File: build.gradle [hunk 2]
```diff
@@ -10,15 +10,20 @@
 
 buildscript {
     repositories {
-    mavenCentral()
-    maven {
-        url "https://oss.sonatype.org/content/repositories/snapshots"
-        }
+        jcenter()
```
[the remark this answers is not in the document]
Comment (planetf1) [reply build.gradle]: fixed

File: build.gradle [hunk 3]
```diff
@@ -46,12 +51,11 @@ allprojects {
     version = '2.9-SNAPSHOT'
 
     repositories {
-         mavenCentral()
-         // snapshots of egeria core ie -SNAPSHOT
-         maven {
-             url "https://oss.sonatype.org/content/repositories/snapshots"
-             }
-         }
+         jcenter()
```
[the remark this answers is not in the document]
Comment (planetf1) [reply build.gradle]: fixed

File: postgres-connector/build.gradle [hunk 4]
```diff
@@ -4,33 +4,26 @@
...
 }
 
 dependencies {
-            implementation 'org.postgresql:postgresql'
-            implementation 'org.slf4j:slf4j-api'
-            implementation 'org.odpi.egeria:data-manager-client'
-            implementation 'org.odpi.egeria:data-manager-api'
-            implementation 'org.odpi.egeria:database-integrator-api'
-            implementation 'org.odpi.egeria:open-connector-framework'
+            implementation 'org.postgresql:postgresql:latest.integration'
```
[the remark this answers is not in the document]
Comment (planetf1) [reply postgres-connector/build.gradle:12]: this regression is still present

File: postgres-connector/src/main/java/org/odpi/openmetadata/adapters/connectors/integration/postgres/PostgresSourceDatabase.java [hunk 5]
```diff
@@ -86,9 +86,9 @@ private String getDatabaseInstance( ) throws SQLException
 
      */
 
-    public List<PostgresDatabase> getDabaseNames( ) throws SQLException, ClassNotFoundException
+    public List<PostgresDatabase> getDabases( ) throws SQLException
```
Comment (planetf1) [inline postgres-connector/src/main/java/org/odpi/openmetadata/adapters/connectors/integration/postgres/PostgresSourceDatabase.java:89]: is getDebases intentional?

File: postgres-connector/src/main/java/org/odpi/openmetadata/adapters/connectors/integration/postgres/PostgresDatabaseConnector.java [hunk 6]
```diff
@@ -94,1239 +105,1768 @@ public void refresh() throws ConnectorCheckedException
...
+
     /**
-     * Checks if any databases need to be removed from egeria
+     * Trawls through a database updating a database where necessary
      *
-     * @param dbs              a list of the bean properties of a Postgres Database
-     * @param knownDatabases    a list of the Databases already known to egeria
-     * @throws                  ConnectorCheckedException
+     * @param postgresDatabase the bean properties of a Postgres Database
+     * @param egeriaDatabase   the egeria database
```
[the remark this answers is not in the document]
Comment (planetf1) [reply postgres-connector/src/main/java/org/odpi/openmetadata/adapters/connectors/integration/postgres/PostgresDatabaseConnector.java]: still present

File: postgres-connector/src/main/java/org/odpi/openmetadata/adapters/connectors/integration/postgres/PostgresDatabaseConnector.java [hunk 7]
```diff
@@ -94,1239 +105,1768 @@ public void refresh() throws ConnectorCheckedException
...
+                    methodName, error);
 
         }
         catch (Exception error)
         {
             if (this.auditLog != null)
             {
                 auditLog.logException(methodName,
-                        PostgresConnectorAuditCode.UNEXPECTTED_ERROR.getMessageDefinition(),
+                        PostgresConnectorAuditCode.UNEXPECTTED_ERROR.getMessageDefinition(methodName, error.getClass().getName(), error.getMessage()),
```
[the remark this answers is not in the document]
Comment (planetf1) [reply postgres-connector/src/main/java/org/odpi/openmetadata/adapters/connectors/integration/postgres/PostgresDatabaseConnector.java]: fixed

File: postgres-connector/src/main/java/org/odpi/openmetadata/adapters/connectors/integration/postgres/ffdc/PostgresConnectorAuditCode.java [hunk 8]
```diff
@@ -20,92 +20,45 @@
...
-            "Ensure that the source Postgres database is available.",
-            "Verify that the source Postgres database is available."),
+    CONNECTOR_CHECKED_EXCEPTION("POSTGRES-CONNECTOR-0005",
+            OMRSAuditLogRecordSeverity.SHUTDOWN,
+            "Method {0} received a connector checked exception from the OMAS server. {1} {2}",
+            "Ensure that the OMAS server is availabale and is responsive",
+            "Check exception details to rectify the problem"),
 
-    UNEXPECTTED_ERROR("POSTGRES-CONNECTOR-0009",
+    UNEXPECTTED_ERROR("POSTGRES-CONNECTOR-0006",
```
[the remark this answers is not in the document]
Comment (planetf1) [reply postgres-connector/src/main/java/org/odpi/openmetadata/adapters/connectors/integration/postgres/ffdc/PostgresConnectorAuditCode.java]: fixed

File: build.gradle (hunk 1 above)
Comment (planetf1) [reply build.gradle]: fixed (albeit with note about replaceable variables above)

File: postgres-connector/src/main/java/org/odpi/openmetadata/adapters/connectors/integration/postgres/PostgresDatabaseConnector.java [hunk 9]
```diff
@@ -94,1239 +105,1980 @@ public void refresh() throws ConnectorCheckedException
...
+                    methodName, error);
 
         }
         catch (UserNotAuthorizedException error)
         {
             if (this.auditLog != null)
             {
                 auditLog.logException(methodName,
-                        PostgresConnectorAuditCode.USER_NOT_AUTORIZED.getMessageDefinition(),
+                        PostgresConnectorAuditCode.USER_NOT_AUTORIZED_EXCEPTION.getMessageDefinition(methodName, error.getClass().getName(), error.getMessage()),
```
Comment (planetf1) [inline postgres-connector/src/main/java/org/odpi/openmetadata/adapters/connectors/integration/postgres/PostgresDatabaseConnector.java:140]: Still have a typo here in NOT_AUTORIZED

**Review by planetf1 — APPROVED** [review]
There are a few outstanding issues
 - A minor typo in naming I think we can correct
 - A minor issue with capitalization
 - A regression in handling of dependency versions
 
 Rather than go through the loop again I am going to merge, and I'll raise a followon issue to address the remaining review comments
  -

**Pull request merged**